cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationService.java

```diff
@@ -828,10 +828,17 @@ public void stopCluster(boolean disableKnoxAutorestart) throws CloudbreakExcepti
         try {
             LOGGER.debug("Stopping all Cloudera Runtime services");
 
-            if (clouderaManagerPollingServiceProvider.checkCmStatus(stack, apiClient).isTimeout()) {
-                skipStopWithStoppedCm();
-            } else {
+            ExtendedPollingResult extendedPollingResult = clouderaManagerPollingServiceProvider.checkCmStatus(stack, apiClient);
+            if (extendedPollingResult.isSuccess()) {
                 stopWithRunningCm(disableKnoxAutorestart, cluster, clustersResourceApi);
+            } else {
+                String errorMessage = "Unknown error from poller";
+                if (extendedPollingResult.getException() != null) {
+                    errorMessage = extendedPollingResult.getException().getMessage();
+                }
+                LOGGER.info("We will skip the CM stop, because the polling result wasn't success: {}, error: {}", extendedPollingResult.getPollingResult(),
+                        errorMessage, extendedPollingResult.getException());
+                skipStopWithStoppedCm();
```

Comment:
nit: could you pls extract this logic to a private method? thanks

Reply:
done